snapshot/policy/scheduling_policy.go

```diff
@@ -147,22 +181,34 @@ func (p *SchedulingPolicy) NextSnapshotTime(previousSnapshotTime, now time.Time)
 		}
 	}
 
-	if ok && p.checkMissedSnapshot(now, previousSnapshotTime, nextSnapshotTime) {
-		// if RunMissed is set and last run was missed, and next run is at least 30 mins from now, then run now
-		nextSnapshotTime = now
-		ok = true
-	}
-
 	return nextSnapshotTime, ok
 }
 
 // Check if a previous snapshot was missed and should be started now.
 func (p *SchedulingPolicy) checkMissedSnapshot(now, previousSnapshotTime, nextSnapshotTime time.Time) bool {
-	const oneDay = 24 * time.Hour
-
 	const halfhour = 30 * time.Minute
 
-	return (len(p.TimesOfDay) > 0 || len(p.Cron) > 0) && p.RunMissed && previousSnapshotTime.Add(oneDay-halfhour).Before(now) && nextSnapshotTime.After(now.Add(halfhour))
+	if !p.RunMissed {
+		return false
+	}
+
+	nextSnapshot := nextSnapshotTime
+	todSnapshot, todOk := p.getNextTimeOfDaySnapshot(previousSnapshotTime.Add(time.Second))
```

Comment:
why do we need to add `time.Second`? Is that really needed?

Reply:
It's a small thing, but I'd suggest extracting `previousSnapshotTime.Add(time.Second)` into its own variable (`momentAfterSnapshot` or something :) )